service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/HearingUrgencyChecker.java

```diff
@@ -26,11 +28,41 @@ public boolean isStarted(CaseData caseData) {
         }
 
         return anyNonEmpty(
-                hearing.getTimeFrame(),
-                hearing.getType(),
-                hearing.getWithoutNotice(),
-                hearing.getReducedNotice(),
-                hearing.getRespondentsAware());
+            hearing.getTimeFrame(),
+            hearing.getType(),
+            hearing.getWithoutNotice(),
+            hearing.getReducedNotice(),
+            hearing.getRespondentsAware());
     }
 
+    @Override
+    public boolean isCompleted(CaseData caseData) {
+        final Hearing hearing = caseData.getHearing();
+
+        if (hearing == null || anyEmpty(
+            hearing.getTimeFrame(),
+            hearing.getType(),
+            hearing.getTypeGiveReason(),
+            hearing.getWithoutNotice(),
+            hearing.getReducedNotice(),
+            hearing.getRespondentsAware())) {
+            return false;
+        }
+
+        if (!hearing.getTimeFrame().equals("Within 18 days")
```

Comment:
When you select a time frame such as "Within 7 days" then the reason will be stored under "reason7Days" instead of "reason" in the case data JSON. Selecting "Same day" will have the reason stored under "Reason". 

This behaviour is kind of annoying for trying to check if the user has entered a reason. Should the behaviour be changed so the reason is always stored under "Reason"?